Components/LearningCore/Source/gov/sandia/cognition/statistics/SummaryStatistics.java

```diff
@@ -27,188 +24,62 @@
  *
  * @author jdwendt
  */
-public class SummaryStatistics
+public interface SummaryStatistics
 {
-
-    /**
-     * The number of entries entered into this engine
-     */
-    private int numEntries;
-
-    /**
-     * The minimum value seen thus far
-     */
-    private double min;
-
-    /**
-     * The maximum value seen thus far
-     */
-    private double max;
-
-    /**
-     * The arithmetic mean of the values seen thus far
-     */
-    private double mean;
-
-    /**
-     * The intermediate value which is optimized for adding new values or
-     * returning actual variance
-     */
-    private double intermediateForVariance;
-
-    /**
-     * Initializes an empty summary. Values returned from this before any data
-     * is entered will not be good (max values, NaNs, etc.)
-     */
-    public SummaryStatistics()
-    {
-        numEntries = 0;
-        min = Double.MAX_VALUE;
-        max = -Double.MIN_VALUE;
-        mean = Double.NaN;
-        intermediateForVariance = Double.NaN;
-    }
-
     /**
      * Add the input value to the summary statistics
      *
      * @param x the value to add to the summary statistics
      */
-    @PublicationReference(type = PublicationType.WebPage, title
-        = "Algorithms for calculating variance - Online algorithm", year = 2016,
-        url
-        = "https://en.wikipedia.org/wiki/Algorithms_for_calculating_variance#Online_algorithm",
-        author = "Wikipedia")
-    public void addValue(double x)
-    {
-        if (Double.isNaN(mean))
-        {
-            mean = 0;
-            intermediateForVariance = 0;
-        }
-        min = Math.min(x, min);
-        max = Math.max(x, max);
-        ++numEntries;
-        double delta = x - mean;
-        mean += delta / numEntries;
-        intermediateForVariance += delta * (x - mean);
-    }
+    public void addValue(double x);
```

Comment:
We should normalize the names here with the other stats classes and such with respect to addValue and merge.